### ISSUE TITLE
Divine Knockout (DKO) (1294660)

### ISSUE THREAD

**Issue by LiamDawe** [ValveSoftware/Proton#6384]
# Compatibility Report
- Name of the game with compatibility issues: Divine Knockout (DKO)
- Steam AppID of the game: 1294660

## System Information
- GPU: Steam Deck
- Driver/LLVM version: SteamOS 3.4 Preview
- Proton version: Any

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

Log file: 
[steam-1294660.log](https://github.com/ValveSoftware/Proton/files/10210171/steam-1294660.log)

## Symptoms 

Easy Anti-Cheat is broken on SteamOS 3.4 for this game, but works on SteamOS 3.3 Stable.

## Reproduction

Try it on SteamOS 3.4, see it fail.

**Comment by waspennator** [comment]
I'm assuming unless something changed, the game is broken now that 3.4 is on stable now.

**Comment by ruineka** [comment]
> I'm assuming unless something changed, the game is broken now that 3.4 is on stable now.

Yup it only worked with older glibc. It's now impossible to play this on the Steam Deck. :(

**Comment by KerrWasHere** [comment]
Is this fixible

**Comment by wipatten** [comment]
Is there any workaround?

**Comment by swilliams-dev** [comment]
I wish the anti-cheat steps from here were automated for developers, might result in less lag time between os and game patches:
https://partner.steamgames.com/doc/steamdeck/proton

Can also confirm, was able to play before the update

**Comment by kamek-pf** [comment]
Personally, I was never able to run the game without the `noeac` flag, and they disabled it with the latest game update. I don't think they ever enabled EAC for Proton, and now, until they do, pretty sure there's no workaround.

**Comment by ruineka** [comment]
> Personally, I was never able to run the game without the `noeac` flag, and they disabled it with the latest game update. I don't think they ever enabled EAC for Proton, and now, until they do, pretty sure there's no workaround.

It worked on older distros running an older glibc and SteamOS 3.3, so I think it was indeed working. I never had to use `noeac`. That said there is no `.so` file in the game files..

**Comment by johnbateman** [comment]
Can confirm, latest version of DKO (Build ID: 10200470) works on 3.3.2 for me without need for noeac flag and on 3.4.2 EAC is broken.

**Comment by kisak-valve** [comment]
Easy Anti-Cheat engine not loading in SteamOS 3.4

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6404.
@gmacbm posted on 2022-12-23T01:47:27:

4# Compatibility Report
- Name of the game with compatibility issues:Divine Knock-out
- Steam AppID of the game:1294660

## System Information
- GPU: AMD
- Driver/LLVM version: Steamdeck
- Kernel version: 
- Proton version:7.0

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [ ] that I have checked whether there are updates for my system available.

## Symptoms:
The dko game easy anti cheat engine using steamdeck is working fine with steam os 3.3 version using proton but upon upgrading to steam os 3.4 with proton the easy anti cheat engine is not loading anymore which prevent the game in launching.


## Reproduction

1.  `Divine Knock out DKO` the AppID is 1294660
2. Linux version: Steam os 3.4
3. Steamdeck device
4. Easy anti cheat engine prevent the game from launching.

**Comment by befabry** [comment]
> Personally, I was never able to run the game without the `noeac` flag, and they disabled it with the latest game update. I don't think they ever enabled EAC for Proton, and now, until they do, pretty sure there's no workaround.

I can confirm, it never worked for me on Arch. No matter the kernel.

It "worked" for my girlfriend on Nobara, but we could never join a game together so she had to use the `-noeac` flag as well. It never worked

**Comment by kamek-pf** [comment]
To follow up on this, it seems @ruineka was right and DKO apparently uses a version of EAC that depends on an older version of glibc.

The [glibc-eac](https://github.com/Frogging-Family/glibc-eac) package allows Arch users to rebuild glibc while excluding commits that break compatibility with older versions of EAC.
The same workaround should work with any distro.

This fixes it for me, I can run the game normally, without the `noeac` flag.

Edit:
DKO seems to be using the EOS version of EAC, which explains why there are no .dll of .so EAC files. EOS EAC appears to be generally broken with glibc 2.36. 
In any case, this issue is probably closely related to #6051.

**Comment by CosmicFusion** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/6384#issuecomment-1367515127

not ubuntu, this patch causes a kernel panic on boot

**Comment by waspennator** [comment]
Was this fixed on deck recently? Noticed it got marked as deck playable.

**Comment by gmacbm** [comment]
Hi valve Software/Proton

     Good day can you please advise how to run this game EAC hence its tag
playable in steamdeck.

Thank you.

**Comment by marioapardo** [comment]
I just tried again from steam deck with proton 7.0.5, and it works, apparently they fixed the EAC error with some update.
